### PR TITLE
Remove PGP signature verification skip for DEV builds

### LINKED ISSUE #3589: Remove pgp verification skip for dev builds
**Describe the enhancement:**

Remove PGP verification skip for dev builds 

**Describe a specific use case for the enhancement or feature:**

**What is the definition of done?**

When the Elastic Agent is built with `DEV=true` the PGP verification happens and there is no way to through the build process produce an agent that skips the PGP verification

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Remove PGP signature verification skip for DEV builds

## Why is it important?

It's not necessary anymore and prevents the PGP signature verification being skipped on a production build if it is ever, by mistake, produced with `DEV=true`.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## How to test this PR locally

Produce a build with `DEV=true`, upgrade (actually downgrade it) it, check there is a log like `Verification with PGP[%d] successful|faied`

## Related issues

- Closes #3589 

## Logs
TODO

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Comment by AndersonQ** [comment]
buildkite test this

File: docs/pgp-sign-verify-artifact.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# Signing Elastic Agent artifacts
+
+This doc covers generating a key, exporting the public key, signing a file and verifying it using GPG as well as pure Go.
```
Comment (michalpristas) [inline docs/pgp-sign-verify-artifact.md:3]: ❤️

**Review by michalpristas — COMMENTED** [review]
havent tested this yet, but it looks good. i like the refactor.
the only downside of this as an issue is losing the abilty to test upgrading to locally built snapshot, but with the use of newly added flags we will not losing it entirely.

**Comment by AndersonQ** [comment]
> havent tested this yet, but it looks good. i like the refactor. the only downside of this as an issue is losing the abilty to test upgrading to locally built snapshot, but with the use of newly added flags with not losing it entirely.

Yes, it's an known issue. The alternative is as you said, skip the verification with the cli flag or sing the local build and pass in the public pgp key. That's also one of the reasons I added the how-to doc

**Review by michalpristas — APPROVED** [review]
seems to work

**Pull request merged**